gremlin-go/driver/performance/performanceSuite.go

```diff
@@ -0,0 +1,465 @@
+			retrieveBytesArr := stats.retrieveBytesArr[1 : runs-1]
+			for i := range retrieveBytesArr {
+				totalBytes += retrieveBytesArr[i]
+			}
+			printBytes(retrieveBytesArr, totalBytes, runs-2, "RETRIEVAL")
+		}
+	}
+}
+
+func printDuration(stats []float64, total float64, runs int, runType string) {
+	fmt.Printf("%s STATS:\t average %fs \t median %fs \t p90 %fs \t p95 %fs \t max %fs \t min %fs \n",
+		runType, total/float64(runs), stats[int(math.Ceil(float64(runs/2)))],
+		stats[int(math.Ceil(float64(runs*90/100)))-1], stats[int(math.Ceil(float64(runs*95/100)))-1], stats[runs-1], stats[0])
+}
+
+func printAllocs(stats []uint64, total uint64, runs int, runType string) {
+	fmt.Printf("%s STATS:\t average %d allocs \t median %d allocs \t p90 %d allocs \t p95 %d allocs \t max %d allocs \t min %d allocs \n",
+		runType, total/uint64(runs), stats[runs/2],
+		stats[int(math.Ceil(float64(runs*90/100)))-1], stats[int(math.Ceil(float64(runs*95/100)))-1], stats[runs-1], stats[0])
+}
+
+func printBytes(stats []uint64, total uint64, runs int, runType string) {
+	fmt.Printf("%s STATS:\t average %d B \t median %d B \t p90 %d B \t p95 %d B \t max %d B \t min %d B \n",
+		runType, total/uint64(runs), stats[runs/2],
+		stats[int(math.Ceil(float64(runs*90/100)))-1], stats[int(math.Ceil(float64(runs*95/100)))-1], stats[runs-1], stats[0])
+}
+
+func getTraversal(testName string, g *GraphTraversalSource) *GraphTraversal {
+	if traversalFcn, ok := traversalMap[testName]; ok {
+		return traversalFcn(g)
+	} else {
+		return nil
+	}
+}
+
+// connection setting constants
+
+const Host = "localhost"
+const Port = 45940
+const GremlinErrors = gremlingo.Error
+const gratefulGraphAlias = "ggrateful"
+const threshold = 4 // same as default
+const writeBufferSize = 1000000
+const readBufferSize = 1000000
+
+//
+// createConnection: Creates a connection to a remote endpoint and returns a
+// GraphTraversalSource that can be used to submit Gremlin queries.
+//
+func createConnection(host string, port, poolSize int) (*GraphTraversalSource, *DriverRemoteConnection, error) {
+	var g *GraphTraversalSource
+	var drc *DriverRemoteConnection
+	var err error
+
+	endpoint := fmt.Sprintf("ws://%s:%d/gremlin", host, port)
+	log.Println("Attempting to connect to : " + endpoint)
+
+	// Establish a new connection and catch any errors that may occur
+	drc, err = gremlingo.NewDriverRemoteConnection(endpoint, func(settings *DriverRemoteConnectionSettings) {
+		settings.LogVerbosity = GremlinErrors
+		settings.TraversalSource = gratefulGraphAlias
+		settings.MaximumConcurrentConnections = poolSize
+		settings.NewConnectionThreshold = threshold
+		settings.WriteBufferSize = writeBufferSize
+		settings.ReadBufferSize = readBufferSize
+	})
+
+	if err != nil {
+		log.Fatalln(err)
+	} else {
+		g = gremlingo.Traversal_().WithRemote(drc)
+	}
+	return g, drc, err
+}
+
+//
+// main: Program entry point. Create the connection, run some tests, shutdown.
+//
+func main() {
```

Comment:
Can we add documentation on how to use it for future needs?

Reply:
Yes, I will add a section to the documentation after everything is finalized.

Reply:
@xiazcy is this done?

Reply:
It is now added as usage info in the program. Issuing `-h` or `--help` displays how to use it, i.e. `go run performanceSuite.go -h`